lib/logging.js

```diff
@@ -0,0 +1,56 @@
+
+'use strict';
+
+var path = require('path');
+var eachModule = require('each-module');
+var winston = require('winston');
+
+module.exports = function (config) {
+	// User-defined configurations take priority, but fallback to defaults if all seem invalid
+	var getArrayOfValidModulesByDirName = function (finalDir) {
+		var modules = [];
+		var modulePusher = function (moduleName, moduleExports, file) {
+			if (typeof moduleExports === 'function') {
+				modules.push(moduleExports);
+			} else {
+				console.warn('Invalid module dropped ' + file);
```

Comment:
Good q, my thought was to try to get the logging configured whilst being defensive as reasonable, so if it does need to throw it can throw in a way more in line with user expectations.  Bit of a chicken & egg...